numpy/lib/format.py

```diff
@@ -599,14 +600,14 @@ def _read_array_header(fp, version):
     if not isinstance(d, dict):
         msg = "Header is not a dictionary: {!r}"
         raise ValueError(msg.format(d))
-    keys = sorted(d.keys())
-    if keys != ['descr', 'fortran_order', 'shape']:
+
+    if EXPECTED_KEYS != d.keys():
         msg = "Header does not contain the correct keys: {!r}"
-        raise ValueError(msg.format(keys))
+        raise ValueError(msg.format(d.keys()))
```

Comment:
Removing the `sorted` above looks correct to me. But I think `d.keys()` now prints slightly less nice?  OTOH, this is de-facto an internal error so I am OK with that.

Reply:
Added the `sorted` bash when generating the error :)